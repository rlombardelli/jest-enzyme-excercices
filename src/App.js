import React, { Component } from "react";
import Demo from "./components/Demo";

import "./App.css";

class App extends Component {
  state = {
    numbers: [1, 2]
  };

  increaseNumbers = () => {
    const newNumbers = this.state.numbers.map(elt => elt + 1);
    this.setState({ numbers: newNumbers });
  };

  addNewNumberToState = () => {
    const newNumber = Math.floor(Math.random() * Math.floor(8)) + 1;
    this.setState({ numbers: [...this.state.numbers, newNumber] });
  };

  render() {
    return (
      <div className="App">
        <h1>Demo time !</h1>
        <span>Current state: {this.state.numbers.toString()}</span>
        <br />
        <Demo numbers={this.state.numbers} />
        <br />
        <button onClick={this.increaseNumbers}>INCREASE BY ONE ALL NUMBERS IN THE STATE</button>
        <br />
        <button onClick={this.addNewNumberToState}>
          ADD A NEW RANDOM NUMBER(1-9) TO THE STATE
        </button>
      </div>
    );
  }
}

export default App;
