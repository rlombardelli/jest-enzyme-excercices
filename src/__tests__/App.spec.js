import React from "react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "../App";

Enzyme.configure({ adapter: new Adapter() });

it("should render the App component", () => {
  const tree = shallow(<App />);

  expect(tree).toMatchSnapshot();
});
