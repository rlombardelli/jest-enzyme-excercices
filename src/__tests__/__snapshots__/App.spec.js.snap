// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the App component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Demo time !
        </h1>,
        <span>
          Current state: 
          1,2
        </span>,
        <br />,
        <Demo
          numbers={
            Array [
              1,
              2,
            ]
          }
        />,
        <br />,
        <button
          onClick={[Function]}
        >
          INCREASE BY ONE ALL NUMBERS IN THE STATE
        </button>,
        <br />,
        <button
          onClick={[Function]}
        >
          ADD A NEW RANDOM NUMBER(1-9) TO THE STATE
        </button>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Demo time !",
        },
        "ref": null,
        "rendered": "Demo time !",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Current state: ",
            "1,2",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Current state: ",
          "1,2",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "numbers": Array [
            1,
            2,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "INCREASE BY ONE ALL NUMBERS IN THE STATE",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "INCREASE BY ONE ALL NUMBERS IN THE STATE",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ADD A NEW RANDOM NUMBER(1-9) TO THE STATE",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "ADD A NEW RANDOM NUMBER(1-9) TO THE STATE",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Demo time !
          </h1>,
          <span>
            Current state: 
            1,2
          </span>,
          <br />,
          <Demo
            numbers={
              Array [
                1,
                2,
              ]
            }
          />,
          <br />,
          <button
            onClick={[Function]}
          >
            INCREASE BY ONE ALL NUMBERS IN THE STATE
          </button>,
          <br />,
          <button
            onClick={[Function]}
          >
            ADD A NEW RANDOM NUMBER(1-9) TO THE STATE
          </button>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Demo time !",
          },
          "ref": null,
          "rendered": "Demo time !",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Current state: ",
              "1,2",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Current state: ",
            "1,2",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "numbers": Array [
              1,
              2,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "INCREASE BY ONE ALL NUMBERS IN THE STATE",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "INCREASE BY ONE ALL NUMBERS IN THE STATE",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ADD A NEW RANDOM NUMBER(1-9) TO THE STATE",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "ADD A NEW RANDOM NUMBER(1-9) TO THE STATE",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
